metomi/rose/suite_engine_procs/cylc.py

```diff
@@ -65,35 +67,60 @@ def get_suite_dir_rel(cls, suite_name, *paths):
         """
         return os.path.join(cls.SUITE_DIR_REL_ROOT, suite_name, *paths)
 
-    def get_suite_jobs_auths(self, suite_name, cycle_name_tuples=None):
-        """Return remote ["[user@]host", ...] for submitted jobs."""
-        # TODO: reimplement for Cylc8?
-        # https://github.com/metomi/rose/issues/2445
-        self.handle_event(
-            Exception(
-                'WARNING: Rose cannot currently inspect the platform a Cylc'
-                ' task has or will run on.\n'
-                ' https://github.com/metomi/rose/issues/2445'
-            )
-        )
-        return []
-
-    def get_task_auth(self, suite_name, task_name):
-        """Return [user@]host for a remote task in a suite.
-
-        Or None if task does not run remotely.
+    def get_suite_jobs_auths(
+        self, suite_name: str, cycle_name_tuples: Tuple[Any] = None
+    ) -> List[str]:
+        """Get hosts of jobs from a Cylc workflow database.
 
+        returns: list of hostname strings.
         """
-        # TODO: reimplement for Cylc8?
-        # https://github.com/metomi/rose/issues/2445
-        self.handle_event(
-            Exception(
-                'WARNING: Rose cannot currently inspect the platform a Cylc'
-                ' task has or will run on.\n'
-                ' https://github.com/metomi/rose/issues/2445'
-            )
+        # n.b. Imports inside function to avoid dependency on Cylc and
+        # Cylc-Rose is Rose is being used with a different workflow engine.
+        from cylc.flow.platforms import get_host_from_platform
+        from cylc.rose.platform_utils import (  # type:ignore
+            get_platforms_from_task_jobs
         )
-        return None
+
+        task_platforms = {}
+        if cycle_name_tuples is not None:
+            for cycle, name in cycle_name_tuples:
+                new_platforms = get_platforms_from_task_jobs(suite_name, cycle)
+                task_platforms[cycle] = new_platforms
+
+        # For each platform get a list of hosts.
+        hosts = []
+        for cycle, tasks in task_platforms.items():
+            for platform in tasks.values():
```

Comment:
This could request multiple hosts from the same platform:

```suggestion
            for platform in str(tasks.values()):
```